### PR TITLE
Feature plant cart id cookie

### PR DESCRIPTION
**Pull request by isapind**
# Plant a cartId cookie when a new cart is started

**Ticket:** [Plant a cartId cookie when a new cart is started](https://trello.com/c/a2qGHNRT/83-plant-a-cartid-cookie-when-a-new-cart-is-started)

## Description

When an order post request has been successful, the returning id is used to plant a cookie.
I have created a cookies file in the utils folder, and an order type to use this order id.

### Screenshots
Screenshot 2022-05-23 at 14 13 44
Screenshot 2022-05-23 at 14 29 20

## Effected Areas

ProductTile.tsx
utils/cookies.ts
types/order.ts

## Developer Checklist

I have

- [x] Made the minimum amount of change required to achieve my goal to a high standard
- [ ] Added tests that describe the change
- [ ] Created tickets for any tech debt incurred

### CODE REVIEW

File: src/services/product.service.ts [hunk 1]
```diff
@@ -35,8 +36,7 @@ export const getSearchResults = async (
 
 export const postOrder = async (
   products: PostProductOrder[]
-): Promise<Product[]> => {
+): Promise<Order> => {
   const { data } = await axios.post(API_POST_ORDER, { products });
-  console.log(data);
   return data;
 };
```
Comment (Si3rr4wow) [inline src/services/product.service.ts:42]: I know this found it's way into this file in an earlier PR but product.services.ts is for products, there should be a separate one for orders

File: src/components/product-tile/ProductTile.tsx [hunk 2]
```diff
@@ -49,15 +50,23 @@ const ProductTileContainer = styled.li`
...
   const handleClick = () => {
-    // TODO: Add to cart
-    const productOrder = [{ product: product.id, quantity: 1 }];
-    postOrder(productOrder);
-    
+    postOrderCreateCookie(product);
+
     const productInCart = cart.find((item) => item.id === product.id);
     productInCart ? changeQuantity(product.id, 1) : addItem(product, 1);
   };
```
Comment (Si3rr4wow) [inline src/components/product-tile/ProductTile.tsx:72]: all the logic in here and `postOrderCreateCookie ` itself should all be in the context. there's nothing special about the product tile or this particular button so don't make it responsible for special things.

**Review by Si3rr4wow — APPROVED** [review]
looks good, just one suggestion about readability

**Review by KhalidStonehouse2001 — APPROVED** [review]
Great✅

**Pull request merged**